src/AppInstallerRepositoryCore/Microsoft/PreIndexedPackageSourceFactory.cpp

```diff
@@ -361,15 +361,40 @@ namespace AppInstaller::Repository::Microsoft
                 }
 
                 std::filesystem::path packageLocation = GetStatePathFromDetails(m_details);
-                packageLocation /= s_PreIndexedPackageSourceFactory_IndexFileName;
+                packageLocation /= s_PreIndexedPackageSourceFactory_PackageFileName;
 
                 if (!std::filesystem::exists(packageLocation))
                 {
                     AICLI_LOG(Repo, Info, << "Data not found at " << packageLocation);
                     THROW_HR(APPINSTALLER_CLI_ERROR_SOURCE_DATA_MISSING);
                 }
 
-                SQLiteIndex index = SQLiteIndex::Open(packageLocation.u8string(), SQLiteIndex::OpenDisposition::Read);
+                // Put a write exclusive lock on the index package.
+                auto indexPackageLock = Utility::ManagedFile::OpenWriteLockedFile(packageLocation, 0);
+
+                // Validate index package trust info.
+                THROW_HR_IF(APPINSTALLER_CLI_ERROR_SOURCE_DATA_INTEGRITY_FAILURE, !Msix::ValidateMsixTrustInfo(packageLocation, WI_IsFlagSet(m_details.TrustLevel, SourceTrustLevel::StoreOrigin)));
+
+                // Create a temp lock exclusive index file.
+                GUID guid;
+                THROW_IF_FAILED(CoCreateGuid(&guid));
+                WCHAR tempFileName[256];
+                THROW_HR_IF(E_UNEXPECTED, StringFromGUID2(guid, tempFileName, ARRAYSIZE(tempFileName)) == 0);
+                auto tempIndexFilePath = Runtime::GetPathTo(Runtime::PathName::Temp);

```

Comment:
I would either create a new function `GetNewTempFilePath` or `GetPathTo(PathName::NewTempFile)`, but both would return the same type of path you are constructing here.  This is definitely reusable.

Reply:
by looking at GetPathTo() code, it's more of returning a directory caller can work on. I'll create GetNewTempFilePath() then